dependencies.py

```diff
@@ -31,21 +31,18 @@ def extra_requirements() -> typing.Dict[str, typing.List[str]]:
     #     - We have a copy of these in extras-requirements.txt. If you modify these, make sure to change it
     #       there as well
     extras_require = {
-        # from 1.17.107 boto3 requires botocore>=1.20.107,<1.21.0 which
-        # conflicts with s3fs 2021.8.1 that has aiobotocore~=1.4.0
-        # which so far (1.4.1) has botocore>=1.20.106,<1.20.107
-        # boto3 1.17.106 has botocore>=1.20.106,<1.21.0, so we must add botocore explicitly
         "s3": [
-            "boto3~=1.9, <1.17.107",
-            "botocore>=1.20.106,<1.20.107",
-            "aiobotocore~=1.4.0",
-            "s3fs~=2021.8.1",
+            "boto3~=1.20.24",
+            # "botocore>=1.20.106,<1.20.107",
+            # "botocore~=1.23.24",
+            "aiobotocore~=2.1.0",
+            "s3fs~=2022.2.0",
         ],
         "azure-blob-storage": [
             "msrest~=0.6.21",
             "azure-core~=1.24",
             "azure-storage-blob~=12.13",
-            "adlfs~=2021.8.1",
+            "adlfs~=2022.2.0",
```

Comment:
worth explaining why changing `s3fs` leads to changing `adlfs` as well (hint: because of `fsspec` itself